### PR TITLE
Removed caching subscription status from plugin

### LINKED ISSUE #19345: Incorrect caching subscription status in newsletter plugin
### Preconditions

1. Magento 2.4-develop

### Steps to reproduce

I spot the issue during placing order with two different community modules installed. In short, during placing order the customer object is subscribed by custom code which utilizes newsletter module, then customer object is saved a couple times, and it results with subscribing him, unsubscribing and subscribing again.

It is a bit complicated to prepare such environment and to reproduce it so I prepared short steps which show incorrect framework behavior.

1. Load customer object using repository and check his subscription status:

    $customer = $customerRepository->getById($customerId);
    $extensionAttributes = $customer->getExtensionAttributes();
    var_dump($extensionAttributes->getIsSubscribed()); // returns false

2. Subscribe customer to newsletter:

    $subscriber = $subscriberFactory->create();
    $subscriber->subscribeCustomerById($customerId);

3. Check subscription status again:

    $customer = $customerRepository->getById($customerId);
    $extensionAttributes = $customer->getExtensionAttributes();
    var_dump($extensionAttributes->getIsSubscribed()); // still returns false

### Expected result

1. getIsSubscribed returns true when runs after subscribing customer

### Actual result

1. getIsSubscribed returns false

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#19345
2. 
### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run Static Tests

**Review by gabrieldagama — APPROVED** [review]
I believe the plugin should not be responsible for "caching", especially because we can't access and clear this cache properly when handling the entity with a repository for example.

We will analyze if there is any performance impact with such changes.

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8817](https://jira.corp.magento.com/browse/ENGCOM-8817) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Load customer object using repository and check his subscription status:
```php
    $customer = $customerRepository->getById($customerId);
    $extensionAttributes = $customer->getExtensionAttributes();
    var_dump($extensionAttributes->getIsSubscribed()); // returns false
```
2. Subscribe customer to newsletter:
```php
    $subscriber = $subscriberFactory->create();
    $subscriber->subscribeCustomerById($customerId);
```
3. Check subscription status again:
```php
    $customer = $customerRepository->getById($customerId);
    $extensionAttributes = $customer->getExtensionAttributes();
    var_dump($extensionAttributes->getIsSubscribed()); // still returns false
```

**Before:** :heavy_multiplication_x: getIsSubscribed returns:
```php
false
```
**After:** :heavy_check_mark: getIsSubscribed returns:
```php
true
```

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Such changes are causing considerable performance degradation, we should review it and provide another solution.

**Comment by gabrieldagama** [comment]
```
Login (Catalog Browsing By Customer) - Another one: 43.6% (mainline = 39 | team = 56) -> DEGRADATION
Load Customer Private Data (Catalog Browsing By Customer): 11.1% (mainline = 45 | team = 50) -> DEGRADATION
Login (Add to Wishlist) - Another one: 47.4% (mainline = 38 | team = 56) -> DEGRADATION
Load Customer Private Data (Add to Wishlist): 13.3% (mainline = 45 | team = 51) -> DEGRADATION
Login (Checkout By Customer) - Another one: 47.4% (mainline = 38 | team = 56) -> DEGRADATION
Load Customer Private Data (Checkout By Customer): 13.3% (mainline = 45 | team = 51) -> DEGRADATION
Simple Product 1 Add To Cart (Checkout By Customer): 35.9% (mainline = 78 | team = 106) -> DEGRADATION
Simple Product 2 Add To Cart (Checkout By Customer): 33.3% (mainline = 90 | team = 120) -> DEGRADATION
Configurable Product 1 Add To Cart (Checkout By Customer): 25.2% (mainline = 119 | team = 149) -> DEGRADATION
Checkout start (Checkout By Customer): 13.8% (mainline = 87 | team = 99) -> DEGRADATION
Checkout Billing/Shipping Information (Checkout By Customer): 12.3% (mainline = 106 | team = 119) -> DEGRADATION
Login (Product Review By Customer) - Another one: 47.4% (mainline = 38 | team = 56) -> DEGRADATION
Checkout success (Checkout By Customer): 11.8% (mainline = 34 | team = 38) -> DEGRADATION
Login (Product Review By Customer) - Another one: 47.4% (mainline = 38 | team = 56) -> DEGRADATION
Load Customer Private Data (Product Review By Customer): 13.3% (mainline = 45 | team = 51) -> DEGRADATION
Login (Add To Cart By Customer) - Another one: 47.4% (mainline = 38 | team = 56) -> DEGRADATION
Load Customer Private Data (Add To Cart By Customer): 13.3% (mainline = 45 | team = 51) -> DEGRADATION
Simple Product 1 Add To Cart (Add To Cart By Customer): 35.9% (mainline = 78 | team = 106) -> DEGRADATION
Simple Product 2 Add To Cart (Add To Cart By Customer): 33.3% (mainline = 90 | team = 120) -> DEGRADATION
Configurable Product 1 Add To Cart (Add To Cart By Customer): 25.2% (mainline = 119 | team = 149) -> DEGRADATION
Remove item (Add To Cart By Customer): 37% (mainline = 81 | team = 111) -> DEGRADATION
Remove item (Add To Cart By Customer) - Another one - Another one: 38.5% (mainline = 78 | team = 108) -> DEGRADATION
Login (Account management) - Another one: 47.4% (mainline = 38 | team = 56) -> DEGRADATION
Load Customer Private Data (Account management): 13.3% (mainline = 45 | team = 51) -> DEGRADATION
Create customer (API Create Customer): 15% (mainline = 40 | team = 46) -> DEGRADATION
Edit Customer (Admin Customer Management): 14.7% (mainline = 95 | team = 109) -> DEGRADATION
Customer Save (Admin Customer Management): 13.6% (mainline = 44 | team = 50) -> DEGRADATION
Create customer (Catalog GraphQL): 17.1% (mainline = 35 | team = 41) -> DEGRADATION
```

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please review and fix integration tests failures

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@ihor-sviziev updated, please review.

File: dev/tests/integration/testsuite/Magento/Newsletter/Controller/Manage/SaveTest.php [hunk 1]
```diff
@@ -126,6 +125,7 @@ public function testUnsubscribeSubscribedCustomer(): void
     private function dispatchSaveAction(bool $isSubscribed): void
     {
         $this->_objectManager->removeSharedInstance(CustomerPlugin::class);
```
Comment (ihor-sviziev) [inline dev/tests/integration/testsuite/Magento/Newsletter/Controller/Manage/SaveTest.php:127]: shouldn't we remove the shared instance also here?
```suggestion
```

Comment (engcom-Foxtrot) [reply dev/tests/integration/testsuite/Magento/Newsletter/Controller/Manage/SaveTest.php:127]: No, the test will fail.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8817](https://jira.corp.magento.com/browse/ENGCOM-8817) has been created to process this Pull Request

**Pull request merged**